custom_hostname_test.go

```diff
@@ -311,6 +316,70 @@ func TestCustomHostname_CustomHostname(t *testing.T) {
 	}
 }
 
+func TestCustomHostname_CustomHostname_WithSSLError(t *testing.T) {
+	setup()
+	defer teardown()
+
+	mux.HandleFunc("/zones/foo/custom_hostnames/bar", func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, "GET", r.Method, "Expected method 'GET', got %s", r.Method)
+
+		w.Header().Set("content-type", "application/json")
+		fmt.Fprintf(w, `{
+"success": true,
+"result": {
+	"id": "bar",
+	"hostname": "foo.bar.com",
+	"ssl": {
+		"type": "dv",
+		"method": "cname",
+		"status": "pending_validation",
+		"cname_target": "dcv.digicert.com",
+		"cname": "810b7d5f01154524b961ba0cd578acc2.foo.bar.com",
+		"validation_errors": {
+			"message": "SERVFAIL looking up CAA for foo.bar.com"
+		}
+	},
+	"status": "pending",
+	"verification_errors": [
+		"None of the A or AAAA records are owned by this account and the pre-generated ownership verification token was not found."
+	],
+	"ownership_verification_http": {
+		"http_url": "http://custom.test.com/.well-known/cf-custom-hostname-challenge/0d89c70d-ad9f-4843-b99f-6cc0252067e9",
```

Comment:
Let's use the reserved example.com domain here as we don't own test.com

```suggestion
		"http_url": "http://example.com/.well-known/cf-custom-hostname-challenge/0d89c70d-ad9f-4843-b99f-6cc0252067e9",
```